### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by neheb**
Mostly nullptr deref.

Found with Coverity.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/util/upnp_quirks.cc [hunk 1]
```diff
@@ -40,7 +40,7 @@
 Quirks::Quirks(const UpnpXMLBuilder& xmlBuilder, const ClientManager& clients, const std::shared_ptr<GrbNet>& addr, const std::string& userAgent)
     : xmlBuilder(xmlBuilder)
 {
-    if (addr || !userAgent.empty())
+    if (addr && !userAgent.empty())
```
Comment (KarlStraussberger) [inline src/util/upnp_quirks.cc:43]: That's breaking things. getInfo has to be called if either `addr` is set or `userAgent`

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
Fine but the quirks change is breaking!

**Comment by neheb** [comment]
Fixed.

File: src/content/content_manager.cc [hunk 2]
```diff
@@ -768,7 +772,7 @@ void ContentManager::_rescanDirectory(const std::shared_ptr<AutoscanDirectory>&
             continue;
         }
 
-        if (shutdownFlag || (task && !task->isValid()))
+        if (shutdownFlag || !task->isValid())
```
Comment (KarlStraussberger) [inline src/content/content_manager.cc:775]: `task` may be unset

File: src/database/search_handler.cc [hunk 3]
```diff
@@ -677,7 +677,7 @@ std::string SortParser::parse(std::string& addColumns, std::string& addJoin)
         } else {
             log_warning("Unknown sort direction '{}' in '{}'", seg, sortCrit);
         }
-        if (!colMapper || !colMapper->mapQuotedList(sort, seg, (desc ? "DESC" : "ASC"))) {
+        if (!colMapper->mapQuotedList(sort, seg, (desc ? "DESC" : "ASC"))) {
```
Comment (KarlStraussberger) [inline src/database/search_handler.cc:680]: can you add a check for `colMapper` and `metaMapper` in the begin of the method
```
if (sortCrit.empty() || !colMapper || !metaMapper)
```

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
New changes - new remarks

File: src/content/content_manager.cc (hunk 2 above)
Comment (neheb) [reply src/content/content_manager.cc]: Fixed.

File: src/database/search_handler.cc (hunk 3 above)
Comment (neheb) [reply src/database/search_handler.cc:680]: Fixed.

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
Found one more issue

File: src/web/edit_load.cc [hunk 4]
```diff
@@ -165,6 +165,8 @@ void Web::EditLoad::process()
     xml2JsonHints->setFieldType("rawvalue", "string");
 
     auto objItem = std::dynamic_pointer_cast<CdsItem>(obj);
+    if (!objItem)
+        return;
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/web/edit_load.cc]: Moved down. I think this should do it.

[the remark this answers is not in the document]
Comment (KarlStraussberger) [reply src/web/edit_load.cc]: no, there is still container code after it. Anyway, `obj` is used before.

[the remark this answers is not in the document]
Comment (neheb) [reply src/web/edit_load.cc]: removed.

[the remark this answers is not in the document]
Comment (KarlStraussberger) [reply src/web/edit_load.cc]: Thank you

**Pull request merged**